Mathlib/Topology/Subbasis.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2025 Fangming Li. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li
+-/
+import Mathlib.Topology.Bases
+import Mathlib.Topology.Compactness.Compact
+
+/-!
+# Results on subbases of topological spaces
+
+In this file, we prove results when some topology is given a subbasis.
+
+## Main results
+
+* `alexander_subbasis hTS`. If `X` has a topology `T` and `hTS : T = generateFrom S`, then `X` is
+  compact if any open cover of `X` with elements in `S` has a finite subcover.
+-/
+
+open TopologicalSpace
+
+variable {X : Type*} [T : TopologicalSpace X] {S : Set (Set X)}
+
+lemma TopologicalSpace.exists_subset_finte_of_mem_nhds_of_subbasis
+    (hTS : T = generateFrom S) {x : X} {n : Set X} (hnx : n ∈ nhds x) :
+    ∃ U ⊆ S, U.Finite ∧ x ∈ ⋂₀ U ∧ ⋂₀ U ⊆ n := by
+  obtain ⟨s, hs, hxs, hsn⟩ := (IsTopologicalBasis.mem_nhds_iff <|
+    isTopologicalBasis_of_subbasis hTS).1 hnx
+  obtain ⟨V, ⟨hV, hVS⟩, hVs⟩ := (Set.mem_image _ _ _).1 hs
+  refine ⟨V, hVS, hV, ?_⟩
+  · rw [hVs]
+    exact ⟨hxs, hsn⟩
+
+lemma Filter.exists_mem_not_mem_of_not_le_nhds_of_subbasis
+    (hTS : T = generateFrom S) {F : Filter X} {x : X} (hFx : ¬F ≤ nhds x) :
+    ∃ s ∈ S, x ∈ s ∧ s ∉ F := by
+  simp only [Filter.le_def, not_forall] at hFx
+  obtain ⟨n, hnx, hnF⟩ := hFx
+  obtain ⟨U, hUS, hU, hxU, hUn⟩ := exists_subset_finte_of_mem_nhds_of_subbasis hTS hnx
+  have h2 : ⋂₀ U ∉ F := fun hUF => hnF <| F.sets_of_superset hUF hUn
+  have h3 : ∃ s ∈ U, s ∉ F := by
+    by_contra hUF
+    have : U ⊆ F.sets := by simpa only [not_exists, not_and, not_not] using hUF
+    exact h2 <| (Filter.sInter_mem hU).2 this
+  obtain ⟨s, hsU, hsF⟩ := h3
+  exact ⟨s, hUS hsU, hxU s hsU, hsF⟩
+
+/--
+The Alexander Subbasis Theorem. If `X` is a topological space with a subbasis `S`, then `X` is
+compact if any open cover of `X` with elements in `S` has a finite subcover.
+-/
+theorem alexander_subbasis (hTS : T = generateFrom S) :
+    (∀ P ⊆ S, ⋃₀ P = ⊤ → ∃ Q ⊆ P, Q.Finite ∧ ⋃₀ Q = ⊤) → CompactSpace X := by
```

Comment:
```suggestion
    (∀ P ⊆ S, ⋃₀ P = .univ → ∃ Q ⊆ P, Q.Finite ∧ ⋃₀ Q = .univ) → CompactSpace X := by
```
The spelling for set is `Set.univ` instead of top.